sunbeam-python/sunbeam/features/vault/feature.py

```diff
@@ -656,8 +656,18 @@ def enable_cmd(
     @pass_method_obj
     def disable_cmd(self, deployment: Deployment, show_hints: bool) -> None:
         """Disable Vault."""
-        self.disable_feature(deployment, show_hints)
         client = deployment.get_client()
+        try:
```

Comment:
The feature should have something like this https://github.com/canonical/snap-openstack/blob/main/sunbeam-python/sunbeam/features/secrets/feature.py#L23

This will take care of checks of dependency features during enable/disable